src/framework/components/joint/component.js

```diff
@@ -28,253 +28,1020 @@ const properties = [
+        return this._angularSpringX;
+    }
+
+    /**
+     * Enables or disables the spring behavior for rotation around the Y-axis.
+     *
+     * @type {boolean}
+     */
+    set angularSpringY(value) {
+        if (this._angularSpringY !== value) {
+            this._angularSpringY = value;
+            if (this._constraint) this._constraint.enableSpring(4, value);
+        }
+    }
+
+    /**
+     * Gets whether the spring behavior is enabled for rotation around the Y-axis.
+     *
+     * @type {boolean}
+     */
+    get angularSpringY() {
+        return this._angularSpringY;
+    }
+
+    /**
+     * Enables or disables the spring behavior for rotation around the Z-axis.
+     *
+     * @type {boolean}
+     */
+    set angularSpringZ(value) {
+        if (this._angularSpringZ !== value) {
+            this._angularSpringZ = value;
+            if (this._constraint) this._constraint.enableSpring(5, value);
+        }
+    }
+
+    /**
+     * Gets whether the spring behavior is enabled for rotation around the Z-axis.
+     *
+     * @type {boolean}
+     */
+    get angularSpringZ() {
+        return this._angularSpringZ;
+    }
+
+    /**
+     * Sets the spring stiffness for rotation around the X-axis.
+     *
+     * @type {number}
+     */
+    set angularStiffnessX(value) {
+        if (this._angularStiffnessX !== value) {
+            this._angularStiffnessX = value;
+            if (this._constraint) this._constraint.setStiffness(3, value);
+        }
+    }
+
+    /**
+     * Gets the spring stiffness for rotation around the X-axis.
+     *
+     * @type {number}
+     */
+    get angularStiffnessX() {
+        return this._angularStiffnessX;
+    }
+
+    /**
+     * Sets the spring stiffness for rotation around the Y-axis.
+     *
+     * @type {number}
+     */
+    set angularStiffnessY(value) {
+        if (this._angularStiffnessY !== value) {
+            this._angularStiffnessY = value;
+            if (this._constraint) this._constraint.setStiffness(4, value);
+        }
+    }
+
+    /**
+     * Gets the spring stiffness for rotation around the Y-axis.
+     *
+     * @type {number}
+     */
+    get angularStiffnessY() {
+        return this._angularStiffnessY;
+    }
+
+    /**
+     * Sets the spring stiffness for rotation around the Z-axis.
+     *
+     * @type {number}
+     */
+    set angularStiffnessZ(value) {
+        if (this._angularStiffnessZ !== value) {
+            this._angularStiffnessZ = value;
+            if (this._constraint) this._constraint.setStiffness(5, value);
+        }
+    }
+
+    /**
+     * Gets the spring stiffness for rotation around the Z-axis.
+     *
+     * @type {number}
+     */
+    get angularStiffnessZ() {
+        return this._angularStiffnessZ;
+    }
+
+    /**
+     * Sets the breaking force threshold for the constraint.
+     *
+     * @type {number}
+     */
+    set breakForce(force) {
+        if (this._constraint && this._breakForce !== force) {
```

Comment:
maybe this should follow other setters, where the internal value gets updated even if _constraint is null, and only the setBreakingImpulseThreshold does not get called